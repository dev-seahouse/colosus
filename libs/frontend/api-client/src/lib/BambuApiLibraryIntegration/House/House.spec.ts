import BambuApiLibraryIntegrationHouseApi from './House';

describe('HouseApi', () => {
  const mockRequest = {
    downPaymentYear: 2021,
    country: 'USA',
    region: 'Ohio',
    city: 'Akron',
    houseType: 'One Bed',
    downPaymentPct: 0.3,
    inflationRate: 0.05,
    currentYear: 2020,
    location: '',
    district: '',
    roomType: '',
  };

  describe('calculateHouseGoalAmount', () => {
    const bambuApiLibraryIntegrationHouseApi =
      new BambuApiLibraryIntegrationHouseApi();
    it('should return a valid response', async () => {
      const res =
        await bambuApiLibraryIntegrationHouseApi.calculateHouseGoalAmount(
          mockRequest
        );
      expect(res.data).toMatchSnapshot();
    });
  });
});
