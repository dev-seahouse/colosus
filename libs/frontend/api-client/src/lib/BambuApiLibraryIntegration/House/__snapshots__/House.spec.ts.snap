// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HouseApi > calculateHouseGoalAmount > should return a valid response 1`] = `
{
  "currency": "USD",
  "downPaymentAmt": 89334348.99,
  "houseCostInflationAdj": 297781163.32,
  "yearsToGoal": 6,
}
`;
